### ISSUE TITLE
DialogService dont open same component multiple times

### ISSUE THREAD

**Issue by macedo8** [primefaces/primeng#15384]
### Describe the bug

I'm developing a component that, after opening a dialog, I can minimize and place only the header of the window on the footer of the page. When I try to open a new window from the same previous component, the dialogService open method returns a null ref.

Is there any way it can return the new object?

### Environment

    public show(componentType: Type<any>, configs: DynamicDialogConfig): DynamicDialogRef {
     configs.templates = {
       header: DialogHeaderComponent
     };

     const ref = this.dialogService.open(componentType, configs);  <-- Error: ref is null
    
     const instance = this.dialogService.getInstance(ref);

     this._controller.addDialog({
       ref: ref,
       instance: instance,
       componentRef: this.dialogService.dialogComponentRefMap.get(ref)!,
       minimize: false
     });

    return ref;
   }

### Angular version

17.3.1

### PrimeNG version

17.12.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.19.1

**Comment by grizzlydev-sarl** [comment]
I stumbled on this too. This is because of the new config attribute `duplicate` that was added in september 2023 abfbe67c8d1ff24bcb33c17977dd66ae7217b302
The result is, by default, we can't open multiple dialog using the same component, unless we set `duplicate`to `true` in the subsequent dialogs config. Eg.:
```ts
const first_dialog_ref: DynamicDialogRef = this.dialogService.open(MyDialogComponent, {
  header: "First dialog",
});

first_dialog_ref.onClose.subscribe((dialogResult: any) => {
  console.log('closing first dialog');
});

const second_dialog_ref: DynamicDialogRef = this.dialogService.open(MyDialogComponent, {
  header: "Second dialog won't open and second_dialog_ref will be null",
  // duplicate: false, // default value
});

try {
  second_dialog_ref.onClose.subscribe((dialogResult: any) => {
    console.log('closing second dialog');
  });
}
catch(error: any) {
  console.log('told ya', error);
}

const third_dialog_ref: DynamicDialogRef = this.dialogService.open(MyDialogComponent, {
  header: "Third dialog will open and third_dialog_ref will be populated",
  duplicate: true,
});

third_dialog_ref.onClose.subscribe((dialogResult: any) => {
  console.log('closing third dialog');
});
```
I don't see what problem this `duplicate` config is supposed to solve. I understand that it's to prevent opening two dialogs with the same component, but I don't see the use case. And if there was such a use case where I'd want to prevent a duplication, it should be an explicit config, on the first dialog. Something like `preventOpenDuplicates: false`, like what's available in the ToastComponent.

I'll add that the addition of this `duplicate` config is an undocumented breaking change. (the attribute is documented, yes, but the change was not)

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation and try the latest PrimeNG version(v19). If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
